test/e2e/scheduling/taints.go

```diff
@@ -426,6 +427,9 @@ var _ = SIGDescribe("NoExecuteTaintManager Multiple Pods [Serial]", func() {
 		pod2.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeHostNameLabel}
 		_, err = testutils.RunPodAndGetNodeName(cs, pod2, 2*time.Minute)
 		framework.ExpectNoError(err)
+		// Wait for pods to be running state before eviction happens
+		framework.ExpectNoError(e2epod.WaitForPodRunningInNamespace(cs, pod1))
```

Comment:
The name of `RunPodAndGetNodeName` function is mis-leading. I initially thought we already check that the pods are running, but looks like that function only waits until the pod is scheduled. We should probably rename that function in a separate PR.

Reply:
Agreed, will change it in a another PR. Thanks :)